packages/modelserver-node/src/test/test-helpers.ts

```diff
@@ -0,0 +1,259 @@
+export interface CustomRoute {
+    method: SupportedMethod;
+    path: string;
+    handler: RequestHandler;
+}
+
+export function route(method: SupportedMethod, handler: RequestHandler): CustomRoute;
+export function route(method: SupportedMethod, path: string, handler: RequestHandler): CustomRoute;
+export function route(method: SupportedMethod, path?: string | RequestHandler, handler?: RequestHandler): CustomRoute {
+    if (typeof path === 'function') {
+        handler = path;
+        path = '/';
+    } else if (typeof path === 'undefined') {
+        handler = (req, res, next) => next();
+        path = '/';
+    } else if (!handler) {
+        handler = (req, res, next) => next();
+    }
+
+    return { method, path, handler };
+}
+
+export function provideEndpoint(container: Container, basePath: string, ...routes: CustomRoute[]): void {
+    const provider: RouteProvider = {
+        configureRoutes(routerFactory: RouterFactory) {
+            const router = routerFactory(basePath);
+            routes.forEach(aRoute => router[aRoute.method](aRoute.path, aRoute.handler));
+        }
+    };
+    container.bind(RouteProvider).toConstantValue(provider);
+}
+
+export interface SocketTimeout {
+    clear(): void;
+}
+
+export function socketTimeout(ws: WebSocket, reject: (reason?: any) => void): SocketTimeout {
+    const result = setTimeout(() => {
+        ws.close();
+        reject(new Error('timeout'));
+    }, 1000);
+
+    return {
+        clear: () => {
+            ws.close();
+            clearTimeout(result);
+        }
+    };
+}
+
+export function captureMessage(ws: WebSocket, filter?: (msg: WebSocket.MessageEvent) => boolean): Promise<WebSocket.MessageEvent> {
+    let timeout: SocketTimeout;
+
+    return new Promise<WebSocket.MessageEvent>((resolve, reject) => {
+        ws.onmessage = msg => {
+            if (!filter || filter(msg)) {
+                resolve(msg);
+            }
+        };
+        timeout = socketTimeout(ws, reject);
+    }).then(result => {
+        timeout.clear();
+        return result;
+    });
+}
+
+export function awaitClosed(transaction: EditTransaction): Promise<boolean> {
+    let timeout: SocketTimeout;
+
+    return new Promise(resolve => {
+        const check = setInterval(() => {
+            if (!transaction.isOpen()) {
+                clearInterval(check);
+                timeout.clear();
+                resolve(true);
+            }
+        }, 50);
+
+        timeout = socketTimeout(transaction['socket'], reason => {
+            clearInterval(check);
+            resolve(false);
+        });
+    });
+}
+
+export function requireArray(owner: object, propertyName: string): unknown[] {
+    expect(owner[propertyName]).to.be.an('array').that.is.not.empty;
+    return owner[propertyName] as unknown[];
+}
+
+export function _assumeThat(this: Context, condition: boolean, reason: string): void {
```

Comment:
If this is going to be an exported test API, it shouldn't start with an underscore.
```suggestion
export function assumeThat(this: Context, condition: boolean, reason: string): void {
```

Reply:
Good catch, agreed!